packages/core/src/lib/actions/callback/oauth/callback.ts

```diff
@@ -207,7 +207,7 @@ async function getUserAndAccount(
         ...tokens,
         provider: provider.id,
         type: provider.type,
-        providerAccountId: user.id.toString(),
+        providerAccountId: userFromProfile.id?.toString()!,
```

Comment:
```suggestion
        providerAccountId: userFromProfile.id,
```

The `ProfileCallback` returns a `User` which expects the `id` to be string already.

Reply:
Yeah good catch, the `id` was typed `string | undefined` though so we added a fallback to `randomUUID()` :+1: